force-app/main/default/classes/RD2_ChangeHistory.cls

```diff
@@ -0,0 +1,210 @@
+public with sharing class RD2_ChangeHistory {
+
+    private RecurringDonationChangeHistory__c record;
+
+    public RD2_ChangeHistory(RecurringDonationChangeHistory__c record) {
+        this.record = record;
+    }
+
+    public RD2_ChangeView getChangeView() {
+        RD2_ChangeView view = new RD2_ChangeView();
+        view.changeType = this.record.ChangeType__c;
+        view.id = this.record.Id;
+
+        if (this.record.NewStatusReason__c != null) {
+            view.statusReason = this.record.NewStatusReason__c;
+        }
+
+        view.changeDate = this.record.EffectiveDate__c;
+        view.fields = buildFieldChangeViews();
+
+        return view;
+    }
+
+    private RD2_FieldChangeView getCampaignChange() {
```

Comment:
Not a blocker, but I would like to pursue a more generic approach to checking for changes and permissions (we chatted about this during our dev call)

Reply:
I gave this a run-through on the code I currently have. I ended up with a long method that invokes `new FieldChangeViewDefinition(field, field, label, fieldType, displayRule);` 7 times to build up the list of definitions for the fields that we want to render. Feels a bit like too much is going on in one spot with that change.